pennylane/transforms/qcut.py

```diff
@@ -428,3 +429,320 @@ def contract_tensors(
     kwargs = {} if use_opt_einsum else {"like": tensors[0]}
 
     return contract(eqn, *tensors, **kwargs)
+
+
+@dataclass()
+class CutStrategy:
+    """
+    A circuit-cutting distribution policy for executing (large) circuits on available (comparably
+    smaller) devices.
+
+    Args:
+        devices (Union[qml.Device, Sequence[qml.Device]]): Single, or Sequence of, device(s).
+            Optional only when ``max_free_wires`` is provided.
+        max_free_wires (int): Number of wires for the largest available device. Optional only when
+            ``devices`` is provided where it defaults to the maximum number of wires among
+            ``devices``.
+        min_free_wires (int): Number of wires for the smallest available device.
+            Optional, defaults to ``max_device_wires``.
+        max_fragments_probed (Union[int, Sequence[int]]): Single, or 2-Sequence of, number(s)
```

Comment:
The original idea is to also allow a non-consecutive list for the probed `k`s. But at the end I opted to interpret lists only as range bounds in this initial implementation because I'm not entirely sure if non-consecutive list of `k`s are really needed; and I want to leave some room in the future for such changes.

As for removing the `int` option, user would then have to provide say `(2, 3)` in order to pick just `2`, which is slightly cumbersome.